scripts/install-mender.sh

```diff
@@ -152,21 +195,82 @@ add_repo() {
     echo "$repo" > /etc/apt/sources.list.d/mender.list
 }
 
-do_install() {
+do_install_open() {
+    # Filter out commercial components
+    local selected_components_open=""
+    for c in $SELECTED_COMPONENTS
+    do
+        if ! echo "$COMMERCIAL_COMPONENTS" | egrep -q "(^| )$c( |\$)"; then
+            selected_components_open="$selected_components_open $c"
+        fi
+    done
+
+    # Return if no open source components selected
+    if [ -z "$selected_components_open" ]; then
+        return
+    fi
+
+    echo "  Installing open source components from APT repository"
+
     apt-get update
     apt-get install -y \
        -o Dpkg::Options::="--force-confdef" \
        -o Dpkg::Options::="--force-confold" \
-       $SELECTED_COMPONENTS
+       $selected_components_open
+
+    echo "  Success! Please run \`mender setup\` to configure the client."
+}
+
+do_install_commercial() {
+    # Filter commercial components
+    local selected_components_commercial=""
+    for c in $SELECTED_COMPONENTS
+    do
+        if echo "$COMMERCIAL_COMPONENTS" | egrep -q "(^| )$c( |\$)"; then
```

Comment:
this one kills me a bit :)
if we declare the commercial components at the begining as a hash:
```bash
declare -A COMMERCIAL_COMPONENTS_array=([mender-monitor]="1" )
COMMERCIAL_COMPONENTS="${!COMMERCIAL_COMPONENTS_array[@]}"
```
then this can be done with `${COMMERCIAL_COMPONENTS_array[${c}]}`

Reply:
That would work for this check and the one in `do_install_open` (by using opposite logic), but not on the analogous check in `usage` for `DEFAULT_COMPONENTS` (as these are not an array).

For that reason (consistency) and the fact that is pre-existing code, I'll leave as it is.

Reply:
ok, but this is a killer, perhaps let's create a small task to refactor that?

Reply:
Why do you think is that bad? It is just okey for an script that is executed once in a device lifetime, and makes it easier to maintain.

This is the place to discuss and agree, I am not creating a task to refactor something so small and then discuss in the task if we want to do it or not...

Reply:
ok, for the sake of continuing, lets keep it as is. it just seemed so suboptimal. so much CPU power to just check a simple thing: if we selected commercial components.